compiler/src/iree/compiler/Utils/ElementPackingUtils.cpp

```diff
@@ -15,25 +15,19 @@
 #include "mlir/Dialect/Arith/IR/Arith.h"
 #include "mlir/IR/BuiltinTypes.h"
 
-// TODO(lialan): remove cl options once frontend can emit packed i1 tensors.
-llvm::cl::opt<bool> clEnableI1Support(
-    "iree-experimental-packed-i1-storage",
-    llvm::cl::desc(
-        "Experimental feature: force to use packed storage for i1 tensors."
-        "Turning on this option will see i1 tensors as if it has "
-        "#iree_encoding.packed_storage attribute."
-        "This is to allow an alternative way to test the packed storage "
-        "feature before frontend can emit packed i1 tensors."
-        "This option can be dropped once the frontend can emit packed i1 "
-        "tensors."),
-    llvm::cl::init(false));
-
 namespace mlir::iree_compiler {
 
-static bool needToPackSubByteElementBitWidthImpl(unsigned bitWidth,
-                                                 bool isPackedStorage) {
-  // Enable i1 support if requested.
-  if (isPackedStorage && bitWidth == 1) {
+bool needToPackSubByteElements(Type type) {
+  unsigned bitWidth = isa<TensorType>(type)
+                          ? IREE::Util::getTypeBitWidth(
+                                dyn_cast<TensorType>(type).getElementType())
+                          : IREE::Util::getTypeBitWidth(type);
+
+  auto rankedTensorType = llvm::dyn_cast_or_null<RankedTensorType>(type);
+  bool isPackedStorage = rankedTensorType &&
+                         IREE::Encoding::hasPackedStorageAttr(rankedTensorType);
```

Comment:
Use dyn_cast because it is never a null type. Also, I think we can simplify it a bit, like:

```cpp
auto rankedTensorType = dyn_cast...
if (!rankedTensorType) {
  return false;
}
unsigned bitWidth = IREE::Util::getTypeBitWidth(rankedTensorType.getElementType());
if (bitWidth == 1 && IREE::Encoding::hasPackedStorageAttr...
```